### PR TITLE
Improve note of failurePolicy and revise IDC configuration

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
Improve note of failurePolicy and revise IDC configuration

### Check List 

Tests 

 - No code

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
None
```

### CODE REVIEW

File: ci/deploy_tidb_operator_staging.groovy [hunk 1]
```diff
@@ -23,12 +23,16 @@ scheduler:
...
 admissionWebhook:
   create: true
+  replicas: 2
   validation:
     statefulSets: true
     pods: true
     pingcapResources: false
   mutation:
     pingcapResources: true
+  failurePolicy:
```
Comment (DanielZhangQD) [inline ci/deploy_tidb_operator_staging.groovy]: Why change them to `Fail` in IDC where the k8s version is v1.12, which does not match the descriptions in `values.yaml` updated above.

Comment (Yisaer) [reply ci/deploy_tidb_operator_staging.groovy]: IDC is under our control, it's fine to set fail for failurePolicy in this case. For community customer, we still recommend set failurePolicy as Ignore as I mentioned in the note.

**Pull request merged**